src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -18,62 +22,196 @@ public class SecretsController : Controller
+            return new SecretResponseModel(result);
+        }
+        else
+        {
+            throw new UnauthorizedAccessException("You don't have read access");
+        }
+
+        return null;
     }
 
-    [HttpPut("secrets/{id}")]
-    public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
+    [HttpPut("secrets/{organizationId}/{id}")]
+    public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid organizationId, [FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
     {
-        var result = await _updateSecretCommand.UpdateAsync(updateRequest.ToSecret(id));
-        return new SecretResponseModel(result);
+        var userId = _userService.GetProperUserId(User).Value;
+        var secret = updateRequest.ToSecret(id, organizationId);
+
+        if(await userHasWriteAccessToProject(secret))
+        {
+            var result = await _updateSecretCommand.UpdateAsync(secret, userId);
+            return new SecretResponseModel(result);
+        } else 
+        {
+            throw new UnauthorizedAccessException("You don't have read access");
+        }
+
+        return null;
     }
 
-    // TODO Once permissions are setup for Secrets Manager need to enforce them on delete.
-    [HttpPost("secrets/delete")]
-    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids)
+    [HttpPost("secrets/{organizationId}/delete")]
+    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids, [FromRoute] Guid organizationId)
     {
-        var results = await _deleteSecretCommand.DeleteSecrets(ids);
+        var userId = _userService.GetProperUserId(User).Value;
+        var results = await _deleteSecretCommand.DeleteSecrets(ids, userId, organizationId);
         var responses = results.Select(r => new BulkDeleteResponseModel(r.Item1.Id, r.Item2));
         return new ListResponseModel<BulkDeleteResponseModel>(responses);
     }
+
+    public async Task<bool> userHasReadAccessToProject(Guid projectId, Guid organizationId)
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+        var hasAccess = false;
+
+        if(projectId != Guid.Empty)
+        {
+            hasAccess = accessClient switch
+            {
+                AccessClientType.NoAccessCheck => true,
+                AccessClientType.User => await _projectRepository.UserHasReadAccessToProject(projectId, userId),
+                AccessClientType.ServiceAccount => await _projectRepository.ServiceAccountHasReadAccessToProject(projectId, userId), 
+                _ => false,
+            };
+        } 
+        else 
+        {
+            hasAccess = orgAdmin;
```

Comment:
There are a few occurrences of `hasAccess = orgAdmin` in the PR. 

I would recommend we try to stick to handling checking access within the accessClient switch statement. 

The `AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin)` is already factoring in the orgAdmin being true or false. 

In this case specifically, I'm not sure if we would need the `Guid.Empty` check.

```
   public async Task<bool> userHasReadAccessToProject(Guid projectId, Guid organizationId)
    {
        if (projectId == Guid.Empty)
        {
            throw new ArgumentException(null, nameof(projectId));
        }
        
        var userId = _userService.GetProperUserId(User).Value;
        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
        
        
        return  accessClient switch
        {
            AccessClientType.NoAccessCheck => true,
            AccessClientType.User => await _projectRepository.UserHasReadAccessToProject(projectId, userId),
            AccessClientType.ServiceAccount => await _projectRepository.ServiceAccountHasReadAccessToProject(projectId, userId), 
            _ => false,
        };
    }
```

Reply:
Thank youuuu!